qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -13,18 +13,14 @@
 """
 Initialize qubit registers to desired arbitrary state.
 """
-import numpy as np
 
-from qiskit.circuit import QuantumCircuit
-from qiskit.circuit import QuantumRegister
-from qiskit.circuit import Instruction
-from qiskit.circuit import Qubit
-from qiskit.circuit.library.data_preparation import StatePreparation
+from qiskit.circuit.library.data_preparation.initializer import Initialize as NewInitialize
+from qiskit.utils.deprecation import deprecate_func
 
 _EPS = 1e-10  # global variable used to chop very small numbers to zero
 
 
-class Initialize(Instruction):
+class Initialize(NewInitialize):
```

Comment:
That could also work if we remove `import qiskit.extensions` from the `qiskit/__init__.py` and move the monkey-patching of circuit methods to the circuit library.

Reply:
you can also at a `__getattr__` on `QuantumCircuit` that issues the deprecation warnings on the monkeypatching because yeah, we shouldn't be importing a deprecated module ourselves.